File: ProyectoCivitasJava/src/civitas/SorpresaSalirCarcel.java
/**
 * Proyecto Civitas
 * Sergio Hervás Cobo
 * Javier Suárez Lloréns
 * Programación y Diseño Orientado a Objetos
 * 2º B3 - Grado en Ingeniería Informática
 * E.T.S.I. Informática y de Telecomunicación
 * Universidad de Granada
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author sergio
 */
public class SorpresaSalirCarcel extends Sorpresa {

    private MazoSorpresas mazo;

    SorpresaSalirCarcel(MazoSorpresas mazo) {
        super("Salir de carcel");
        this.mazo = mazo;
    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            boolean comprobacion = false;
            informe(actual, todos);
            for (int k = 0; k < todos.size(); k++) {
                if (todos.get(k).tieneSalvoconducto()) {
                    comprobacion = true;
                }
            }
            if (!comprobacion) {
                todos.get(actual).obtenerSalvoconducto(this);
                salirDelMazo();
            }
        }
    }

    void usada() {
        mazo.habilitarCartaEspecial(this);
    }

    void salirDelMazo() {
        mazo.inhabilitarCartaEspecial(this);
    }

    @Override
    public String toString(){
        String salida = "Sorpresa para salir de la cárcel";
        return salida;
    }
    
    
}

File: ProyectoCivitasRuby/lib/modelo/sorpresa_salir_carcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class SorpresaSalirCarcel < Sorpresa
    public_class_method :new 
    def initialize(mazo)
      super("Salir de la carcel")
      @mazo = mazo
    end 
    
    #@Override
    def aplicarajugador(actual, todos)
      if(jugador_correcto(actual, todos))
        informe(actual, todos)
        comprobacion = false
        for k in todos do
          if (k.tiene_salvoconducto)
            comprobacion = true
          end
        end
          
        if(!comprobacion)
          todos[actual].obtener_salvoconducto(self)
          salir_del_mazo
        end
      end
    end
            
    def salir_del_mazo
        @mazo.inhabilitar_carta_especial(self)
    end

    
    def usada
        @mazo.habilitar_carta_especial(self)
    end

    def to_s
      salida = "{@texto}\nSorpresa para salir de la carcel"
    end
     
  end
end